amf-webapi/shared/src/main/scala/amf/plugins/document/webapi/contexts/SpecEmitterContext.scala

```diff
@@ -114,11 +114,25 @@ trait SpecEmitterFactory {
 }
 
 trait TagToReferenceEmitter extends PartEmitter {
-  val target: DomainElement
+  val link: DomainElement
 
-  val label: Option[String]
+  val label: Option[String] = link match {
+    case l: Linkable => l.linkLabel.option()
+    case _           => None
+  }
+
+  val referenceLabel: String = label.getOrElse(link.id)
+}
 
-  val referenceLabel: String = label.getOrElse(target.id)
+object ReferenceEmitterHelper {
+
+  def emitLinkOr(l: DomainElement with Linkable, b: PartBuilder, refs: Seq[BaseUnit] = Nil)(inner: => Unit)(
```

Comment:
Would rename for something easier to understand. "inner" doesn't convey much meaning.